ChildProject/annotations.py

```diff
@@ -712,6 +712,217 @@ def import_annotations(
+
+            else:
+                logger_annotations.warning("File %s already exists. To overwrite, specify parameter ''overwrite_existing''", output_filename)
+                return annotation_result
+
+        # find if there are annotation indexes in the same set that overlap the new annotation
+        # as it is not possible to annotate multiple times the same audio stretch in the same set
+        ovl_annots = self.annotations[(self.annotations['set'] == output_set) &
+                                      (self.annotations[
+                                           'annotation_filename'] != annotation_filename) &  # this condition avoid matching a line that should be overwritten (so has the same annotation_filename), it is dependent on the previous block!!!
+                                      (self.annotations['recording_filename'] == annotation['recording_filename']) &
+                                      (self.annotations['range_onset'] < annotation['range_offset']) &
+                                      (self.annotations['range_offset'] > annotation['range_onset'])
+                                      ]
+        if ovl_annots.shape[0] > 0:
+            array_tup = list(
+                ovl_annots[['set', 'recording_filename', 'range_onset', 'range_offset']].itertuples(index=False,
+                                                                                                    name=None))
+            annotation_result[
+                "error"] = f"derivation for set <{output_set}> recording <{annotation['recording_filename']}> from {annotation['range_onset']} to {annotation['range_offset']} cannot continue because it overlaps with these existing annotation lines: {array_tup}"
+            logger_annotations.error("Error: %s", annotation['error'])
+            # (f"Error: {annotation['error']}")
+            annotation_result["imported_at"] = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+            return annotation_result
+
+        path = os.path.join(
+            self.project.path,
+            "annotations",
+            annotation["set"],
+            "converted", #EXPAND
+            annotation["annotation_filename"],
+        )
+
+        #TODO CHECK FOR DTYPES
+        df_input = pd.read_csv(path)
+        df = None
+
+        try:
+            if callable(import_function):
+                df = import_function(df_input)
+            elif import_function in DERIVATIONS.keys():
+                df = DERIVATIONS[import_function](df_input)
+            else:
+                raise ValueError(
+                    "derivation value '{}' unknown, use one of {}".format(import_function, DERIVATIONS.keys())
+                )
+        except:
+            annotation["error"] = traceback.format_exc()
+            logger_annotations.error("An error occurred while processing '%s'", path, exc_info=True)
+
+        if df is None or not isinstance(df, pd.DataFrame):
```

Comment:
this is a tricky place, we want to check that the function used returned a dataframe as expected, because people could use the python api to call with any function they want